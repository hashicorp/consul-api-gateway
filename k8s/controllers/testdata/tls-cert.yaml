
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: annotations
  annotations:
    "polar.hashicorp.com/consul-http-scheme": "https"
    "polar.hashicorp.com/consul-ca-secret": "super-secret"
spec:
  listeners:  
  - protocol: HTTP
    port: 80
    name: http
    allowedRoutes:
      namespaces:
        from: Same