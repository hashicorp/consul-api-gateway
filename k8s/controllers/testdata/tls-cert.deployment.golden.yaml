metadata:
  creationTimestamp: null
  labels:
    managedBy: polar
    name: polar-annotations
  name: annotations
spec:
  selector:
    matchLabels:
      managedBy: polar
      name: polar-annotations
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        managedBy: polar
        name: polar-annotations
    spec:
      containers:
      - command:
        - /bootstrap/polar
        - exec
        - -log-json
        - -log-level
        - info
        - -gateway-host
        - $(IP)
        - -gateway-ports
        - HTTP:http:80
        - -gateway-name
        - annotations
        - -consul-http-address
        - $(HOST_IP):8500
        - -consul-ca-cert-file
        - /ca/tls.crt
        env:
        - name: IP
          valueFrom:
            fieldRef:
              fieldPath: status.podIP
        - name: HOST_IP
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
        image: envoyproxy/envoy:v1.18-latest
        name: polar
        ports:
        - containerPort: 80
          name: http
          protocol: TCP
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
        - mountPath: /ca
          name: ca
          readOnly: true
      initContainers:
      - command:
        - cp
        - /bin/polar
        - /bootstrap/polar
        image: hashicorp/polar:0.1.0
        name: polar-init
        resources: {}
        volumeMounts:
        - mountPath: /bootstrap
          name: bootstrap
        - mountPath: /certs
          name: certs
        - mountPath: /ca
          name: ca
          readOnly: true
      volumes:
      - emptyDir: {}
        name: bootstrap
      - emptyDir: {}
        name: certs
      - name: ca
        secret:
          secretName: super-secret
status: {}
